File: java/src/main/java/com/hindu/calendar/model/MasaName.java
package com.hindu.calendar.model;

public enum MasaName {
    CHAITRA(1, "Chaitra"),
    VAISHAKHA(2, "Vaishakha"),
    JYESHTHA(3, "Jyeshtha"),
    ASHADHA(4, "Ashadha"),
    SHRAVANA(5, "Shravana"),
    BHADRAPADA(6, "Bhadrapada"),
    ASHVINA(7, "Ashvina"),
    KARTIKA(8, "Kartika"),
    MARGASHIRSHA(9, "Margashirsha"),
    PAUSHA(10, "Pausha"),
    MAGHA(11, "Magha"),
    PHALGUNA(12, "Phalguna");

    private final int number;
    private final String displayName;

    MasaName(int number, String displayName) {
        this.number = number;
        this.displayName = displayName;
    }

    public int number() {
        return number;
    }

    public String displayName() {
        return displayName;
    }

    public static MasaName fromNumber(int num) {
        for (MasaName m : values()) {
            if (m.number == num) return m;
        }
        throw new IllegalArgumentException("Invalid masa number: " + num);
    }
}

File: swift/Sources/HinduCalendar/Model/MasaName.swift
public enum MasaName: Int, CaseIterable {
    case chaitra = 1
    case vaishakha = 2
    case jyeshtha = 3
    case ashadha = 4
    case shravana = 5
    case bhadrapada = 6
    case ashvina = 7
    case kartika = 8
    case margashirsha = 9
    case pausha = 10
    case magha = 11
    case phalguna = 12

    public var displayName: String {
        switch self {
        case .chaitra: return "Chaitra"
        case .vaishakha: return "Vaishakha"
        case .jyeshtha: return "Jyeshtha"
        case .ashadha: return "Ashadha"
        case .shravana: return "Shravana"
        case .bhadrapada: return "Bhadrapada"
        case .ashvina: return "Ashvina"
        case .kartika: return "Kartika"
        case .margashirsha: return "Margashirsha"
        case .pausha: return "Pausha"
        case .magha: return "Magha"
        case .phalguna: return "Phalguna"
        }
    }

    public static func fromNumber(_ num: Int) -> MasaName {
        guard let masa = MasaName(rawValue: num) else {
            fatalError("Invalid masa number: \(num)")
        }
        return masa
    }
}
